Initial import of "Protect an API" scenario (#4)
Azure-Samples/ms-identity-docs-code-java: protect-api/README.md

> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.

### 2. Update code sample with app registration values

Open the [_application.yml_](src/main/resources/application.yml) file and modify the three Azure Active Directory configuration properties using the values from your [app's registration in the Azure portal](https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app).

```yaml
# 'Tenant ID' of your Azure AD instance - this value is a GUID
tenant-id: value-here

# 'Application ID URI' of app registration in Azure portal - this value typically starts with api://
app-id-uri: value-here
```

> ### 3. Install package(s) and compile application

Reviewer: I'm less convinced now that a separate **Install packages** section adds value because so many (most? all?) samples use a single command to install all their dependencies.

The original intent of this section was to provide the install command that included the auth package name, but with `dotnet`, `mvn`, `npm`, and `pip` taking care of it behind the scenes with `install.txt` and similar batch facilities, that intent isn't being realized (which is fine because it's devlang/framework-idiomatic way to go).

Should probably collapse it all into **Run the application**, and we'll (content devs) figure out another way to communicate/highlight the package install command for just the auth librar[y|ies] used.

Author: Yeah, they have been pretty anemic and tiny.  I'm for collapsing them, reducing the vertical scroll on the page a bit.

To install Spring Boot and MSAL libraries:

```bash
mvn install
```

## Run the application

```bash
mvn spring-boot:run
```

> ## Access the api

Reviewer: ```suggestion
## Access the API

```

Author: Whoops, good catch!

> Open postman, curl, or similar and make an HTTP GET request to **http://localhost:8080** with an `Authorization` header of `Bearer {valid-access-token}`. If everything worked, the sample app should produce output similar to this:

Reviewer: ```suggestion
Using Postman, curl, or a similar application, issue an HTTP GET request to *http://localhost:8080* with an `Authorization` header of `Bearer {VALID-ACCESS-TOKEN}`.

For example, if you use curl and everything worked, the sample you should receive a response from the API similar to this:

```

```console

> $ curl http://localhost:8080 -H "Authorization: Bearer {valid-access-token}"

Reviewer: ```suggestion
$ curl http://localhost:8080 -H "Authorization: Bearer {VALID-ACCESS-TOKEN}"
```

Author: I'll make these changes to the OTHER two open PRs as well.

Hello, world. You were able to access this because you provided a valid access token with the Greeting.Read scope as a claim.
```

## About the code

> This Java API uses the Spring Boot web framework. The app has a single route that requires an access token. The access token will be automatically validated by MSAL. A missing or invalid (expired, wrong audience, etc) token will result in a `401` response. A otherwise valid token without the proper scopes will result in a `403` response. A valid token with a proper scope of (`Greeting.Read`) will result in the "Hello, world." message.

Reviewer: ```suggestion
This Java API uses the Spring Boot web framework. The app has a single route that requires an access token. The access token will be automatically validated by MSAL:

- A missing or invalid (expired, wrong audience, etc) token will result in a `401` response.
- An otherwise valid token without the proper scope will result in a `403` response.
- A valid token with the proper scope of `Greeting.Read` will result in the "Hello, world." message.
```

<!-- Keeping yaml frontmatter commented out for now
---
# Metadata required by https://docs.microsoft.com/samples/browse/
# Metadata properties: https://review.docs.microsoft.com/help/contribute/samples/process/onboarding?branch=main#add-metadata-to-readme
languages:
- java
page_type: sample
name: "Java API written in Spring Boot that both protects its own endpoints"

> description: "This Java API that protects it's own endpoints using JWT scope validation. The code in this sample is used by one or more articles on docs.microsoft.com."

Reviewer 2: it's should be its

Author: Indeed!  Good catch @skabou, addressed in https://github.com/Azure-Samples/ms-identity-docs-code-java/pull/4/commits/79b1d634b85d249428c2151503cde0d17cba095f

Author: Marking resolved (@skabou doesn't have permission to do so)

products:
- azure
- azure-active-directory

> - ms-graph

Reviewer 2: Should ms-graph be here?

Author: Good catch @skabou, addressed in https://github.com/Azure-Samples/ms-identity-docs-code-java/pull/4/commits/79b1d634b85d249428c2151503cde0d17cba095f

Author: Marking resolved (@skabou doesn't have permission to do so)
